src/Resource/Collection.js

```diff
@@ -0,0 +1,101 @@
+import ParameterBag from './ParameterBag';
+
+export default class Collection {
+  /**
+   * @param {String} name
+   * @param {HyralRepository} repository
+   */
+  constructor(name, repository) {
+    /** @private */
+    this.name = name;
+
+    /** @private */
+    this.repository = repository;
+
+    /** @private */
+    this.metadata = {
+      loading: false,
+      loaded: false,
+      parameterBag: null,
+      paging: {
+        count: null,
+        pages: null,
+      },
+    };
+
+    /** @private */
+    this.loadingPromise = null;
```

Comment:
Deze variable wordt nog niet gebruikt. 
Ik neem aan dat de bedoeling is om in de load een promise aan te maken die gerolved wordt op het moment dat de data terug komt?

Reply:
mag weg nog even